src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 8;
```

Comment:
why 8?

Reply:
I tried to implement my ArrayList as far as possible from original implementation. Didn't knew it is mandatory.